soroban-auth/src/lib.rs

```diff
@@ -67,10 +76,9 @@ fn verify_account_signatures(env: &Env, auth: &AccountSignatures, name: Symbol,
         }
 
         env.verify_sig_ed25519(&sig.public_key, &msg_bytes, &sig.signature);
-
-        weight = weight
-            .checked_add(acc.signer_weight(&sig.public_key))
-            .expect("weight overflow");
+        // Signature weight can be at most 255, hence overflow isn't possible
+        // here as 255 * MAX_ACCOUNT_SIGNATURES is < u32::MAX.
+        weight += acc.signer_weight(&sig.public_key);
```

Comment:
Once https://github.com/stellar/rs-soroban-env/pull/493#discussion_r984255735 is changed to return 0 from the signer_weight function, we need to update this code to test for `0` and panic.

Reply:
Done.